client/post-editor/editor-post-formats/accordion.jsx

```diff
@@ -1,68 +1,64 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	pick = require( 'lodash/pick' ),
-	find = require( 'lodash/find' ),
-	classNames = require( 'classnames' ),
-	isEmpty = require( 'lodash/isEmpty' );
+import React, { PropTypes } from 'react';
+import { has, isEmpty } from 'lodash';
+import classNames from 'classnames';
 
 /**
  * Internal dependencies
  */
-var Accordion = require( 'components/accordion' ),
-	Gridicon = require( 'components/gridicon' ),
-	siteUtils = require( 'lib/site/utils' ),
-	PostFormats = require( './' );
+import Accordion from 'components/accordion';
+import Gridicon from 'components/gridicon';
+import siteUtils from 'lib/site/utils';
+import PostFormats from './';
 
-module.exports = React.createClass( {
+export default React.createClass( {
 	displayName: 'EditorPostFormatsAccordion',
 
 	propTypes: {
-		site: React.PropTypes.object,
-		post: React.PropTypes.object,
-		postFormats: React.PropTypes.arrayOf( React.PropTypes.shape( {
-			slug: React.PropTypes.string,
-			label: React.PropTypes.string
-		} ) )
```

Comment:
I'm fairly certain I've implemented selectors with both of these return value options in the past, and while it depends on the UI requirements, I'd probably side more with @youknowriad on this one, which is that:

- The reducer shape tracks an object of post formats, keyed by slug
- The selectors transform this to whatever is most appropriate for the UI

I think it's rare that the UI requirements would dictate returning the original object from the selector.

Here, I see two requirements:

- Showing the label of the selected format in the accordion subtitle
- Showing a list of all post format options

To the first, we could achieve this with a selector `getPostFormat( state, siteId, slug )` to retrieve the single post format ~~object~~ label (instead of trying to look it up from the entire set as has been done 'til now, which can be error prone and is nice to have a tested selector for).

To the second, since it's used as a single unit (list), an array seems to be the most appropriate return value from `getPostFormats( state, siteId )`.